Tests / runtime_tests (ubuntu-latest, 3.12)  failing test_aln2basecnt_more

`test_aln2basecnt_more` seems to fail due to numerical accuracy on a new Ubuntu version?

```

________________ test_aln2basecnt_more[stoploss_with_deletions] ________________

tmp_path = PosixPath('/tmp/pytest-of-runner/pytest-0/test_aln2basecnt_more_stoploss1')
combin = 'stoploss_with_deletions'

    @pytest.mark.parametrize(
        "combin", ["stopgain_with_deletions", "stoploss", "stoploss_with_deletions"]
    )
    def test_aln2basecnt_more(tmp_path, combin):
        # data on LFS
        crampath = PurePath("tests/test_frameshift_deletions_checks")
        datapath = PurePath("tests/test_aln2basecnt")
    
        files = {
            "bscnt": f"{combin}.basecnt.tsv",
            "cov": f"{combin}.coverage.tsv",
            "stat": f"{combin}.stats.yaml",
        }
        exp = {f: datapath / f for f in files.values()}  # expected
        out = {f: tmp_path / f for f in files.values()}  # current
    
        subprocess.check_call(
            [
                "aln2basecnt",
                "--first",
                "1",
                "--basecnt",
                out[files["bscnt"]],
                "--coverage",
                out[files["cov"]],
                "--stats",
                out[files["stat"]],
                crampath / f"{combin}.cram",
            ]
        )
    
        # check output
        for f in files.values():
            with open(exp[f], "rt") as expf, open(out[f], "rt") as outf:
>               assert [r for r in expf] == [row for row in outf]
E               AssertionError: assert ['coverage:\n...903.0\n', ...] == ['coverage:\n...903.0\n', ...]
E                 
E                 At index 12 diff: '  std: 592.6809370090612\n' != '  std: 592.6809370090613\n'
E                 
E                 Full diff:
E                   [
E                       'coverage:\n',
E                       '  25%: 50.5\n',
E                       '  50%: 235.0\n',
E                       '  75%: 489.0\n',
E                       '  GC: 0.38709601717158904\n',
E                       '  count: 29903.0\n',
E                       '  insert_mean: 387.21666413834953\n',
E                       '  max: 5903.0\n',
E                       '  mean: 396.1788114904859\n',
E                       '  min: 0.0\n',
E                       '  readlen_mean: 224.6460672026699\n',
E                       '  reads: 52736.0\n',
E                 -     '  std: 592.6809370090613\n',
E                 ?                             ^
E                 +     '  std: 592.6809370090612\n',
E                 ?                             ^
E                       '\n',
E                   ]

tests/test_aln2basecnt.py:103: AssertionError
```

--- a/tests/test_aln2basecnt.py
+++ b/tests/test_aln2basecnt.py
@@ -1,7 +1,9 @@
 import subprocess
 from pathlib import PurePath
 import json
+import yaml
 import pytest
+import numpy as np
 
 
 @pytest.mark.parametrize(
@@ -63,7 +65,20 @@ def test_aln2basecnt(tmp_path, combin):
     # check output
     for f in files.values():
         with open(exp[f], "rt") as expf, open(out[f], "rt") as outf:
-            assert [r for r in expf] == [row for row in outf]
+            if '.yaml' in f:
+                exp_data = yaml.safe_load(expf)
+                out_data = yaml.safe_load(outf)
+                if isinstance(exp_data, dict) and len(exp_data) == 1:
+                    key = list(exp_data.keys())[0]
+                    for k, v in exp_data[key].items():
+                        if isinstance(v, float):
+                            assert np.isclose(v, out_data[key][k], atol=1e-10)
+                        else:
+                            assert v == out_data[key][k]
+                else:
+                    assert exp_data == out_data
+            else:
+                assert [r for r in expf] == [row for row in outf]
 
 
 @pytest.mark.parametrize(
@@ -100,4 +115,17 @@ def test_aln2basecnt_more(tmp_path, combin):
     # check output
     for f in files.values():
         with open(exp[f], "rt") as expf, open(out[f], "rt") as outf:
-            assert [r for r in expf] == [row for row in outf]
+            if '.yaml' in f:
+                exp_data = yaml.safe_load(expf)
+                out_data = yaml.safe_load(outf)
+                if isinstance(exp_data, dict) and len(exp_data) == 1:
+                    key = list(exp_data.keys())[0]
+                    for k, v in exp_data[key].items():
+                        if isinstance(v, float):
+                            assert np.isclose(v, out_data[key][k], atol=1e-10)
+                        else:
+                            assert v == out_data[key][k]
+                else:
+                    assert exp_data == out_data
+            else:
+                assert [r for r in expf] == [row for row in outf]

